weave/trace/util.py

```diff
@@ -168,6 +168,59 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
     return deco
 
 
+KT = TypeVar("KT")
+VT = TypeVar("VT")
+
+
+class InvertableDict(MutableMapping[KT, VT]):
```

Comment:
this is cool... but 2 things:
1. Seems a bit overkill: `def invert_dict(d): return {v: k for k, v in d.items()}`?
2. Would put this in it's own file since it has nothing specific to do with trace

Reply:
1. That's not guaranteed to be a bijection though because `invert_dict(invert_dict(x))` may not equal `x` (consider 2 keys map to 1)
2. Ok, we can move